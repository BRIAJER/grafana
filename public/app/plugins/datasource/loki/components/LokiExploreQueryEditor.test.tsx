import React from 'react';
import { shallow } from 'enzyme';
import LokiExploreQueryEditor from './LokiExploreQueryEditor';
import { LokiDatasource } from '../datasource';
import { LokiQuery } from '../types';
import { ExploreMode, PanelData, LoadingState, dateTime } from '@grafana/data';

const setup = (propOverrides?: object) => {
  const datasourceMock: unknown = {};
  const datasource: LokiDatasource = datasourceMock as LokiDatasource;
  const onRunQuery = jest.fn();
  const onChange = jest.fn();
  const query: LokiQuery = { expr: '', refId: 'A', maxLines: 0 };
  const data: PanelData = {
    state: LoadingState.NotStarted,
    series: [],
    request: {
      requestId: '1',
      dashboardId: 1,
      interval: '1s',
      panelId: 1,
      range: {
        from: dateTime('2020-01-01', 'YYYY-MM-DD'),
        to: dateTime('2020-01-02', 'YYYY-MM-DD'),
        raw: {
          from: dateTime('2020-01-01', 'YYYY-MM-DD'),
          to: dateTime('2020-01-02', 'YYYY-MM-DD'),
        },
      },
      scopedVars: {},
      targets: [],
      timezone: 'GMT',
      app: 'Grafana',
      startTime: 0,
    },
    timeRange: {
      from: dateTime('2020-01-01', 'YYYY-MM-DD'),
      to: dateTime('2020-01-02', 'YYYY-MM-DD'),
      raw: {
        from: dateTime('2020-01-01', 'YYYY-MM-DD'),
        to: dateTime('2020-01-02', 'YYYY-MM-DD'),
      },
    },
  };
  const history: any[] = [];
  const exploreMode: ExploreMode = ExploreMode.Logs;

  const props: any = {
    query,
    data,
    datasource,
    exploreMode,
    history,
    onChange,
    onRunQuery,
  };

  Object.assign(props, propOverrides);

  return shallow(<LokiExploreQueryEditor {...props} />);
};

describe('LokiExploreQueryEditor', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
