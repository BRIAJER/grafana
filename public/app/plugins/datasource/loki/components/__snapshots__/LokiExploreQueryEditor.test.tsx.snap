// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LokiExploreExtraField should render component 1`] = `
<Fragment>
  <Component
    ExtraFieldElement={
      <Memo(LokiExploreExtraField)
        label="Line limit"
        min={0}
        onChangeFunc={[Function]}
        onKeyDownFunc={[Function]}
        type="number"
        value="0"
      />
    }
    absoluteRange={
      Object {
        "from": 1577833200000,
        "to": 1577919600000,
      }
    }
    data={
      Object {
        "request": Object {
          "app": "Grafana",
          "dashboardId": 1,
          "interval": "1s",
          "panelId": 1,
          "range": Object {
            "from": "2019-12-31T23:00:00.000Z",
            "raw": Object {
              "from": "2019-12-31T23:00:00.000Z",
              "to": "2020-01-01T23:00:00.000Z",
            },
            "to": "2020-01-01T23:00:00.000Z",
          },
          "requestId": "1",
          "scopedVars": Object {},
          "startTime": 0,
          "targets": Array [],
          "timezone": "GMT",
        },
        "series": Array [],
        "state": "NotStarted",
        "timeRange": Object {
          "from": "2019-12-31T23:00:00.000Z",
          "raw": Object {
            "from": "2019-12-31T23:00:00.000Z",
            "to": "2020-01-01T23:00:00.000Z",
          },
          "to": "2020-01-01T23:00:00.000Z",
        },
      }
    }
    datasource={Object {}}
    history={Array []}
    logLabelOptions={Array []}
    onChange={[MockFunction]}
    onLabelsRefresh={[Function]}
    onLoadOptions={[Function]}
    onRunQuery={[MockFunction]}
    query={
      Object {
        "expr": "",
        "maxLines": 0,
        "refId": "A",
      }
    }
    syntax={null}
    syntaxLoaded={false}
  />
</Fragment>
`;
